src/tools/perf/lib/libperf.c

```diff
@@ -1087,6 +1088,10 @@ static ucs_status_t ucp_perf_test_alloc_mem(ucx_perf_context_t *perf)
         buffer_size = ucx_perf_get_message_size(params);
     }
 
+    if (ucx_perf_cmd_is_atomic(params->command)) {
+        buffer_size *= 2;
```

Comment:
discussed offline: need to use one buffer, and use the payload to signal the last iteration (e.g put(1), fadd(1), cswap(0 to 1)